platform/MCU/shared/STM32/src/hw_ticks.c

```diff
@@ -39,8 +39,15 @@ static volatile system_tick_t system_millis_clock = 0;
  */
 void System1MsTick(void)
 {
+	int is = __get_PRIMASK();
```

Comment:
Any reason not to use HAL_disable_irq() and HAL_enable_irq() in interrupts_hal.c?